clients/java/src/test/java/com/google/myanmartools/TransliterateTest.java

```diff
@@ -761,4 +768,69 @@ public void u2ZTests() {
     }
   }
 
+  @Test
+  /**
+   * Compatibility test for Z->U conversion using golden file.
+   */
+  public void z2UCompatibilityTest() {
+    String fileInputPath = "src/test/resources/com/google/myanmartools/mmgov_zawgyi_src.txt";
+    String goldenInputPath = "src/test/resources/com/google/myanmartools/mmgov_unicode_out.txt";
+
+    // Open input files.
+    try {
+        BufferedReader input = Files.newBufferedReader(Paths.get(fileInputPath), UTF_8);
+        BufferedReader golden = Files.newBufferedReader(Paths.get(goldenInputPath), UTF_8);
+
+        int lineNum = 0;
+        String zLine  = input.readLine();
+        while (zLine != null) {
+          String convertedLine = z2U.convert(zLine);
+          String goldenLine = golden.readLine();
+          assertWithMessage(
+              "Difference in line " + lineNum + "\n" +
+              "  expected = " + goldenLine + "\n" +
+              "  actual   = " + convertedLine + "\n").
+              that(convertedLine).isEqualTo(goldenLine);
+          zLine  = input.readLine();
+          lineNum ++;
+        }
+    }
+    catch (IOException ex) {
+      org.junit.Assert.fail("** Error in file open/read in z2UCompatibiityTest: " + ex.getMessage());
+    }
+  }
+
+    @Test
+  /**
+   * Compatibility test for U->Z conversion using golden file.
+   * Note that normalization of Zawgyi is needed for comparison.
+   */
+  public void u2ZCompatibilityTest() {
+    String fileInputPath = "src/test/resources/com/google/myanmartools/udhr_mya_unicode_src.txt";
+    String goldenInputPath = "src/test/resources/com/google/myanmartools/udhr_mya_zawgyi_out.txt";
+
+    // Open input files.
+    try {
+        BufferedReader input = Files.newBufferedReader(Paths.get(fileInputPath), UTF_8);
+        BufferedReader golden = Files.newBufferedReader(Paths.get(goldenInputPath), UTF_8);
+
+        int lineNum = 0;
+        String uLine  = input.readLine();
+        while (uLine != null) {
+          String convertedLine = zNorm.convert(u2Z.convert(uLine));
```

Comment:
I don't think zNorm should be required here.  When the golden file was generated, did it go through zNorm?

Reply:
I'm not normalizing the golden output files right now.  I see some benefit to doing that. Perhaps in the next iteration.

Reply:
Let me try to explain my normalization comment more clearly.

Let U2Z be the ICU4J transliterator run on the unicode-to-zawgyi rules, and let U2Z' be the Java conversion rules created by the genconvert project.  Using this notation,

```
compatTestU2Zoutput == U2Z(compatTestU2Zsource)
```

And what we want to test is that

```
compatTestU2Zoutput == U2Z'(compatTestU2Zsource)
```

i.e., that

```
U2Z(compatTestU2Zsource) == U2Z'(compatTestU2Zsource)
```

However, you are actually testing

```
ZNorm'(compatTestU2Zoutput) == ZNorm'(U2Z'(compatTestU2Zsource))
```

i.e.,

```
ZNorm'(U2Z(compatTestU2Zsource)) == ZNorm'(U2Z'(compatTestU2Zsource))
```

This doesn't make sense.  In this context you should be making the stronger comparison that the outputs are equal without passing them through ZNorm'.